src/algebra/group/inj_surj.lean

```diff
@@ -243,6 +243,27 @@ protected def group [group M₁] (f : M₁ → M₂) (hf : surjective f)
 { mul_left_inv := hf.forall.2 $ λ x, by erw [← inv, ← mul, mul_left_inv, one]; refl,
   ..‹has_inv M₂›, .. hf.monoid f one mul }
 
+/-- A type endowed with `1`, `*`, `⁻¹` and `/` is a group,
+if it admits an surjective map from a group that preserves these operations.
+
+This version of `surjective.group` makes sure that the `/` operation is defeq
+to the specified division operator.
+-/
+@[to_additive
+"A type endowed with `0`, `+` and `-` is an additive group,
+if it admits an surjective map from an additive group that preserves these operations.
+
+This version of `surjective.add_group` makes sure that the `-` operation is defeq
+to the specified subtraction operator."]
+protected def group_div [has_div M₂] [group M₁] (f : M₁ → M₂) (hf : surjective f)
+  (one : f 1 = 1) (mul : ∀ x y, f (x * y) = f x * f y) (inv : ∀ x, f (x⁻¹) = (f x)⁻¹)
+  (div : ∀ x y, f (x / y) = f x / f y) :
+  group M₂ :=
+{ div_eq_mul_inv := hf.forall.2 $ λ x, hf.forall.2 $
+    λ y, by erw [← div, div_eq_mul_inv, mul, inv]; refl,
+  mul_left_inv := hf.forall.2 $ λ x, by erw [← inv, ← mul, mul_left_inv, one]; refl,
```

Comment:
Does it make sense to express `group` above in terms of `group_div` to avoid repeating a proof?

Reply:
I saved a line.